--- lshw-B.02.14/src/core/scsi.cc~	2010-09-05 19:36:47.411010711 +0200
+++ lshw-B.02.14/src/core/scsi.cc	2010-09-05 19:02:19.148903622 +0200
@@ -842,7 +842,7 @@
 
           if (!controller)
           {
-            controller = node.addChild(hwNode::hwNode("scsi", hw::storage));
+            controller = node.addChild(hwNode("scsi", hw::storage));
             if (controller)
             {
               controller->setLogicalName(host_logicalname(number));
